lib/runtime/storage/trie.go

```diff
@@ -176,34 +192,41 @@ func (t *TrieState) NextKey(key []byte) []byte {
 	defer t.mtx.RUnlock()
 
 	if currentTx := t.getCurrentTransaction(); currentTx != nil {
-		allEntries := t.state.Entries()
-		maps.Copy(allEntries, currentTx.upserts)
-
-		keys := maps.Keys(allEntries)
-		sort.Strings(keys)
+		allSortedKeys := append(t.sortedKeys, currentTx.sortedKeys...)
+		// Find key position
+		pos, found := slices.BinarySearch(allSortedKeys, string(key))
+		if found {
+			pos = pos + 1
+		}
 
-		for _, k := range keys {
-			if k > string(key) && !currentTx.deletes[k] {
-				return []byte(k)
-			}
+		// Get next key based on that position
+		if pos < len(allSortedKeys) {
+			k := allSortedKeys[pos]
+			return []byte(k)
```

Comment:
I saw that we are sever times doing string(k) when adding to keys and []byte(k)when returning, maybe can jsut make sortedKeys to by array of byte arrays

Reply:
Yes we have a "problem" with the keys because they are `[]bytes` but slices in golang doesn't implement `cmp.Ordered` so we need to use `string` for that.
For example, if I change `sortedKeys` to a `[][]byte` we wont be able to use `slices.BinarySearch` since it expects a `cmp.Ordered` https://pkg.go.dev/slices#BinarySearch
Something similar happens when we need to add the keys to a map, the map keys should be comparable and we need to define the map as `map[string]T` for that 😞